ckanext/ytp/request/tests/test_plugin.py

```diff
@@ -1,44 +1,65 @@
+from ckan.model.system_info import get_system_info
+from ckan.tests.helpers import submit_and_follow
 
-class TestYTPRequestsPlugin(object):
-    '''Tests for the ckanext.example_iauthfunctions.plugin module.
+webtest_submit = helpers.webtest_submit
+PLUGIN_CONTROLLER = 'ckanext.ytp.request.controller:YtpRequestController'
 
-    Specifically tests that overriding parent auth functions will cause
-    child auth functions to use the overridden version.
+class TestViewingActionedReferral(helpers.FunctionalTestBase):
+    '''
+        Test that viewing an already actioned membership request does not 404
     '''
-    @classmethod
-    def setup_class(cls):
-        '''Nose runs this method once to setup our test class.'''
-        # Test code should use CKAN's plugins.load() function to load plugins
-        # to be tested.
-        ckan.plugins.load('ytp_request')
-
-    def teardown(self):
-        '''Nose runs this method after each test method in our test class.'''
-        # Rebuild CKAN's database after each test method, so that each test
-        # method runs with a clean slate.
-        model.repo.rebuild_db()
-
-    @classmethod
-    def teardown_class(cls):
-        '''
-        Nose runs this method once after all the test methods in our class
-        have been run.
-        '''
-        # We have to unload the plugin we loaded, so it doesn't affect any
-        # tests that run after ours.
-        ckan.plugins.unload('ytp_request')
-
-    def sample_test(self):
-        '''
-        Placeholder test.
-        '''
-        pass
+
+    _load_plugins = (u'ytp_request', )
+
+    @mock.patch('ckanext.ytp.request.logic.action.create.flash_success')
+    @mock.patch('ckanext.ytp.request.logic.action.update.flash_success')
+    def test_viewing_actioned_referral(self, mocked_create_flash, mocked_update_flash):
+        app = self._get_test_app()
+
+        # setup
+        org = factories.Organization()
+        sysadmin = factories.Sysadmin()
+        regular_user = factories.User()
+
+        # user creates membership request
+        membership_request = helpers.call_action(
+            'member_request_create',
+            {'user': regular_user['name']},
+            group=org['name'],
+            role='member'
+        )
+
+        # admin approves membership
+        membership_approval = helpers.call_action(
+            'member_request_approve',
+            {'user': sysadmin['name']},
+            approve='approve',
+            mrequest_id=membership_request['id']
+        )
+
+        # admin view membership request page again
+        url = url_for(
+            controller=PLUGIN_CONTROLLER,
+            action='show',
+            mrequest_id=membership_request['id']            
+        )
+        response = app.get(
+            url,
+            extra_environ={'REMOTE_USER': sysadmin['name'].encode('ascii')},
+            status=200
+        )
+
+        # app.get() will raise a "Bad response" error if the page 404s
```

Comment:
Yes, I think you need a proper "assert" section here. It quite unique scenario but the following should do the job:
```
    try:
        app.get(
            url,
            extra_environ={'REMOTE_USER': sysadmin['name'].encode('ascii')},
            status=200
        )
    except BadResponseError:
        self.fail("meaningful test fail message which'll help find out what's wrong in the future")
```
more info (take a look at the most voted comment to this answer): https://stackoverflow.com/a/4319870/3432120